### PR TITLE
Add import warnings instead of constant.

### LINKED ISSUE #373: Improve error when assigning to imported variable
```js
import {snakeSpeed, updateSnake as update, drawSnake as draw, gameover as gameover,length as length, body as body} from "./snake.js"

gameover = true;
```
```
hello.js:3:1: error: assignment to const variable [E003]
hello.js:1:75: note: const variable declared here [E003]
```

quick-lint-js could be more helpful and say "imported variable" instead of "const variable".

`variable_kind` should tell us whether it's an imported variable or a `const`-declared variable.

### PR DESCRIPTION
**Pull request by suryasis-hub**
Fixes #373 : The idea is to reuse the existing kind checks from the switch cases.

### CODE REVIEW

**Comment by suryasis-hub** [comment]
I have read and hereby agree to quick-lint-js' Contributor License Agreement (CLA-v1.md).

**Comment by suryasis-hub** [comment]
recheck

**Comment by suryasis-hub** [comment]
recheck

File: src/quick-lint-js/error.h [hunk 1]
```diff
@@ -62,7 +62,16 @@
       ERROR(QLJS_TRANSLATABLE("assignment to const variable"), assignment)     \
           NOTE(QLJS_TRANSLATABLE("const variable declared here"),              \
                declaration))                                                   \
-                                                                               \
+  QLJS_ERROR_TYPE(                                                             \
+      error_assignment_to_import_variable, "E0185",                            \
```
Comment (strager) [inline src/quick-lint-js/error.h:66]: When I wrote the task, I had in mind that we would reuse E0003 error_assignment_to_const_variable, but customize its message based on `var_kind`.

Making a separate error code seems fine too. :+1:

File: test/test-lint.cpp [hunk 2]
```diff
@@ -1044,11 +1044,11 @@ TEST(test_lint, assign_to_mutable_variable_shadowing_immutable_variable) {
   EXPECT_THAT(v.errors, IsEmpty());
 }
 
-TEST(test_lint, assign_to_immutable_variable) {
+TEST(test_lint, assign_to_const_variable) {
   const char8 declaration[] = u8"x";
   const char8 assignment[] = u8"x";
 
-  for (variable_kind kind : {variable_kind::_const, variable_kind::_import}) {
+  for (variable_kind kind : {variable_kind::_const}) {
```
Comment (strager) [inline test/test-lint.cpp:1051]: :+1:

**Review by strager — APPROVED** [review]
Looks good to me. Let me know when I should merge this.

**Comment by strager** [comment]
Wait, this looks like a copy of @keyehzy's #606. What happened?

**Comment by suryasis-hub** [comment]
@strager , actually I raised the pull request and later realized there is one already doing something similar. I had put in my comments there so that either pull request should work.

**Comment by strager** [comment]
I'll take #606 instead, since it was created first.

**Pull request closed without merging**